internal/api/registry.go

```diff
@@ -43,42 +43,124 @@ var (
 	VersionResourceType      = azcorearm.NewResourceType(ProviderNamespace, "locations/"+VersionResourceTypeName)
 )
 
-type VersionedCreatableResource[T any] interface {
-	Normalize(*T)
-	GetVisibility(path string) (VisibilityFlags, bool)
-	ValidateVisibility(current VersionedCreatableResource[T], updating bool) []arm.CloudErrorBody
+type Resource interface {
```

Comment:
Should this be mixed in to the types?

Reply:
Sorry, "Mixin" is a very OOP term.

AFAIK there is nothing actually using this (I tested this by deleting it and it linted successfully). I would have expected to see it under the "HCPOpenShiftCluster****" structs.

Reply:
Resolved offline